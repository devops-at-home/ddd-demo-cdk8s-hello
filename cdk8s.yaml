language: typescript
app: node main.js
imports:
  - k8s@1.21.0
