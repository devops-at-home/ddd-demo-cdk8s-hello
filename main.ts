import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import {
  KubeDeployment,
  KubeService,
  IntOrString,
  KubeIngress,
} from './imports/k8s';

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const label = { app: 'ddd-demo-cdk8s' };

    const service = new KubeService(this, 'service', {
      spec: {
        type: 'NodePort',
        ports: [{ port: 80, targetPort: IntOrString.fromNumber(8080) }],
        selector: label,
      },
    });

    new KubeIngress(this, 'ingress', {
      metadata: {
        annotations: {
          'kubernetes.io/ingress.class': 'alb',
          'alb.ingress.kubernetes.io/scheme': 'internet-facing',
          'alb.ingress.kubernetes.io/target-type': 'ip',
        },
      },
      spec: {
        rules: [
          {
            http: {
              paths: [
                {
                  path: '/',
                  pathType: 'Prefix',
                  backend: {
                    service: {
                      name: service.name,
                      port: {
                        number: 80,
                      },
                    },
                  },
                },
              ],
            },
          },
        ],
      },
    });

    new KubeDeployment(this, 'deployment', {
      spec: {
        replicas: 2,
        selector: {
          matchLabels: label,
        },
        template: {
          metadata: { labels: label },
          spec: {
            containers: [
              {
                name: 'hello-kubernetes',
                image: 'paulbouwer/hello-kubernetes:1.9',
                ports: [{ containerPort: 8080 }],
                env: [
                  {
                    name: 'MESSAGE',
                    value: 'Hello DDD!',
                  },
                ],
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new MyChart(app, 'ddd-demo-cdk8s');
app.synth();
