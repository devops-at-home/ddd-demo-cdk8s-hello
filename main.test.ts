import { MyChart } from './main';
import { Testing } from 'cdk8s';

describe('DDD demo', () => {
  test('Snapshot', () => {
    const app = Testing.app();
    const chart = new MyChart(app, 'test-chart');
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
