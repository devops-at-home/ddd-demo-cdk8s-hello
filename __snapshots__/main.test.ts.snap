// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DDD demo Snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-chart-service-c8a895ee",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "ddd-demo-cdk8s",
      },
      "type": "NodePort",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": Object {
      "annotations": Object {
        "alb.ingress.kubernetes.io/scheme": "internet-facing",
        "alb.ingress.kubernetes.io/target-type": "ip",
        "kubernetes.io/ingress.class": "alb",
      },
      "name": "test-chart-ingress-c82f94dc",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "service": Object {
                    "name": "test-chart-service-c8a895ee",
                    "port": Object {
                      "number": 80,
                    },
                  },
                },
                "path": "/",
                "pathType": "Prefix",
              },
            ],
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-chart-deployment-c8d4e901",
    },
    "spec": Object {
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "app": "ddd-demo-cdk8s",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "ddd-demo-cdk8s",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "MESSAGE",
                  "value": "Hello DDD!",
                },
              ],
              "image": "paulbouwer/hello-kubernetes:1.9",
              "name": "hello-kubernetes",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
